tests-utils/src/main/scala/tech/beshu/ror/utils/containers/LdapSingleContainer.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.utils.containers
+
+import better.files.File
+import com.dimafeng.testcontainers.SingleContainer
+import monix.execution.Scheduler.Implicits.global
+import org.testcontainers.containers.{GenericContainer as JavaGenericContainer}
+import tech.beshu.ror.utils.containers.LdapSingleContainer.InitScriptSource
+import tech.beshu.ror.utils.containers.windows.{NonStoppableInMemoryLdapService, WindowsPseudoSingleContainerLdap}
+import tech.beshu.ror.utils.misc.OsUtils
+import tech.beshu.ror.utils.misc.ScalaUtils.*
+
+import scala.concurrent.duration.*
+import scala.language.{implicitConversions, postfixOps}
+
+trait LdapSingleContainer extends SingleContainer[JavaGenericContainer[_]] {
```

Comment:
`LdapSingleContainer` is a common interface for both preexisting LdapContainer - testcontainer, and the Windows implementation, that runs in memory.